mambaSharedFramework/HLSValueTypes.swift

```diff
@@ -189,6 +274,109 @@ public struct HLSClosedCaptions: FailableStringLiteralConvertible {
+        /// for alternative Renditions in the same group with compatible attributes and a greater channel count. It MAY
+        /// be dynamically spatialized.
+        case downmix = "DOWNMIX"
+
+        /// Allows `init` without having to allocate a new `String` object.
+        init?(str: Substring) {
+            switch str {
+            case "BINAURAL": self = .binaural
+            case "IMMERSIVE": self = .immersive
+            case "DOWNMIX": self = .downmix
+            default: return nil
+            }
+        }
+    }
+
+    public init?(string: String) {
+        var count: Int?
+        var spatialAudioCodingIdentifiers: [String]?
+        var specialUsageIdentifiers: [SpecialUsageIdentifier]?
+        let enumeratedSplit = string.split(separator: "/").enumerated()
+        for (index, str) in enumeratedSplit {
+            switch index {
+            case 0: count = Self.parseChannelCount(str: str)
+            case 1: spatialAudioCodingIdentifiers = Self.parseSpatialAudioCodingIdentifiers(str: str)
+            case 2:
+                guard let ids = Self.parseSpecialUsageIdentifiers(str: str) else {
+                    // In the case that we don't recognize one of the special usage identifiers, leading to nil being
+                    // parsed out, I believe it is better to fail the entire parsing, as otherwise we could mislead the
+                    // user of the library into thinking that there are less special usage identifiers than there
+                    // actually are in the CHANNELS attribtue.
+                    return nil
+                }
+                specialUsageIdentifiers = ids
+            default: break // In the future there may be more parameters defined.
+            }
+        }
+        // Count is required to have been parsed.
+        guard let count else {
+            return nil
+        }
+        self.count = count
+        self.spatialAudioCodingIdentifiers = spatialAudioCodingIdentifiers ?? []
+        self.specialUsageIdentifiers = specialUsageIdentifiers ?? []
+    }
+
+    public init(
+        count: Int,
+        spatialAudioCodingIdentifiers: [String],
+        specialUsageIdentifiers: [SpecialUsageIdentifier]
+    ) {
+        self.count = count
+        self.spatialAudioCodingIdentifiers = spatialAudioCodingIdentifiers
+        self.specialUsageIdentifiers = specialUsageIdentifiers
+    }
+
+    private static func parseChannelCount(str: Substring) -> Int? {
+        Int(string: String(str))
+    }
+
+    private static func parseSpatialAudioCodingIdentifiers(str: Substring) -> [String] {
+        let split = str.split(separator: ",")
+        var identifiers = [String]()
+        for id in split where id != "-" {
+            identifiers.append(String(id))
+        }
+        return identifiers
+    }
+
+    private static func parseSpecialUsageIdentifiers(str: Substring) -> [SpecialUsageIdentifier]? {
+        let split = str.split(separator: ",")
```

Comment:
I think your current approach is more performant but you could also do
```swift
let res = split.map({ SpecialUsageIdentifier(str: $0)}).compactMap({ $0 })
```
for read-ability

Reply:
The compact map approach would break my early return behavior to throw away the whole struct in case any properties are unrecognized. I can change to this if I took the `unknown(String)` approach I mentioned.

Reply:
Updated here: 02d2cb9f8a36379b6b1d49891da3bae364c1309c